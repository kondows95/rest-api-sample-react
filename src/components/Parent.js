import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom';
import { IntlProvider } from "react-intl";
import { chooseLocale } from '../locations';
import store from '../store';

const Parent = ({children}) => {
  return (
    <Provider store={store}>
      <IntlProvider locale={'en'} messages={chooseLocale('en')}>
        <Router>
          {children}
        </Router>
      </IntlProvider>
    </Provider>
  );
}
export default Parent


