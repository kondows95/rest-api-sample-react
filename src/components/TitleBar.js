import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'

import { Grid, Box, AppBar, IconButton, Toolbar, Badge, Menu, MenuItem, Divider, ListItemIcon, ListItemText  } from '@material-ui/core'
import { Link as RouterLink, withRouter } from 'react-router-dom'
import { 
  Menu as MenuIcon,
  ShoppingCart as ShoppingCartIcon,
  AccountBalanceWallet as LogoutIcon
} from '@material-ui/icons';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import { FormattedMessage } from 'react-intl';

const useStyles = makeStyles(theme => ({
  appBar: {
    //zIndex: theme.zIndex.drawer + 1,
    zIndex: 999
  },
  menuButton: {
    marginRight: theme.spacing(2),
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  rightIcon: {
    marginLeft: 'auto',
    marginRight: theme.spacing(1),
  },
  imgIcon: {
    width: 30,
    height: 30,
    borderRadius: '100%'
  },
  selected: {
    background: '#e9e9ef'
  }
}));

const ITEM_HEIGHT = 60;

const TitleBar = ({locale, setLocale,handleDrawerToggle, totalQuantity, changeAuthState, fetchAuthedUser, signOut, user, history}) => {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  }

  const handleClose = () => {
    setAnchorEl(null);
  }
  
  const handleLocale = (event, locale) => {
    event.preventDefault();
    setLocale(locale);
    setAnchorEl(null);
  }
  
  const isFirstRef = React.useRef(true)
  React.useEffect(() => {
    if (isFirstRef.current) {
      isFirstRef.current = false;
      fetchAuthedUser()
    }
  })

  const handleLogout = event => {
    event.preventDefault();
    signOut();
  }

  return (
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="Open drawer"
          edge="start"
          onClick={handleDrawerToggle}
          disabled={(handleDrawerToggle === null)}
          className={classes.menuButton}
        >
          <MenuIcon />
        </IconButton>
        <Box flexGrow={1} fontSize="h6.fontSize">
          <FormattedMessage id="Top.Title" defualtMessage="Title" />
        </Box>
        <Box mr={0} ml="auto">
          <Grid container>
            <RouterLink to="/cart" style={{ color: "white"}}>
              <IconButton
                color="inherit"
              >
                <Badge
                  color="secondary"
                  badgeContent={totalQuantity}
                  invisible={totalQuantity > 0 ? false : true}>
                  <ShoppingCartIcon className="fa-shopping-cart" />
                </Badge>
              </IconButton>
            </RouterLink>
          
              <IconButton
                color="inherit"
                aria-label="More"
                aria-controls="long-menu"
                aria-haspopup="true"
                onClick={handleClick}
               >
                <MoreVertIcon />
              </IconButton>
              <Menu
                id="long-menu"
                anchorEl={anchorEl}
                elevation={0}
                getContentAnchorEl={null}
                anchorOrigin={{
                  vertical: 'bottom',
                  horizontal: 'center',
                }}
                transformOrigin={{
                  vertical: 'top',
                  horizontal: 'center',
                }}
                keepMounted={false}
                open={open}
                onClose={handleClose}
                PaperProps={{
                  style: {
                    maxHeight: ITEM_HEIGHT * 4.5,
                    width: 200,
                  },
                }}
              >
                
                <MenuItem disabled={true} >
                  <FormattedMessage id="Menu.Title" defualtMessage="Change Language" />
                </MenuItem>
                
                <Divider /> 
                
                <MenuItem className={ locale === 'ja' ? classes.selected : null }  onClick={event => handleLocale(event, 'ja')}>
                  <ListItemIcon>
                    <img src="https://cdn2.iconfinder.com/data/icons/world-flags-1-1/100/Japan-512.png" className={classes.imgIcon} alt='Japanese' />
                  </ListItemIcon>
                  <ListItemText> <FormattedMessage id="Menu.Japanese" defualtMessage="Japanese" /> </ListItemText>
                </MenuItem>
                
                <MenuItem className={ locale === 'en' ? classes.selected : null } onClick={event => handleLocale(event, 'en')}>
                  <ListItemIcon>
                    <img src="https://cdn3.iconfinder.com/data/icons/world-flags-circular-1/512/49-Great_Britain_United_Kingdom_UK_England_Union_Jack_country_flag_-512.png" className={classes.imgIcon} alt='English' />
                  </ListItemIcon>
                  <ListItemText> <FormattedMessage id="Menu.English" defualtMessage="English" /> </ListItemText>
                </MenuItem>
                
                <Divider />
                
                <MenuItem  onClick={handleLogout} >
                  <ListItemIcon>
                    <LogoutIcon />
                  </ListItemIcon>
                  <ListItemText> <FormattedMessage id="Menu.SingOut" defualtMessage="Sing Out" /> </ListItemText>
                </MenuItem>
                
              </Menu>
      
          </Grid>  
        </Box>
      </Toolbar>
    </AppBar>
  )
}

TitleBar.propTypes = {
  handleDrawerToggle: PropTypes.func,
  changeAuthState: PropTypes.func.isRequired,
  fetchAuthedUser: PropTypes.func.isRequired
}

TitleBar.defaultProps = {
  handleDrawerToggle: null,
}

export default withRouter(TitleBar);