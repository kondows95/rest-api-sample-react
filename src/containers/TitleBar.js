import { connect } from 'react-redux';
import TitleBar from '../components/TitleBar';
import { fetchAuthedUser, changeAuthState, signOut } from '../modules/auth';
import { setLocale } from '../modules/locale';

export default connect(
  (state) => ({
    totalQuantity:  state.cart.totalQuantity,
    authState: state.auth.authState,
    dialogOpened: state.auth.dialogOpened,
    user: state.auth.user,
    loading: state.auth.loading,
    locale: state.locale.locale
  }),
  (dispatch) => ({
    fetchAuthedUser: () => dispatch(fetchAuthedUser()),
    changeAuthState: (value) => dispatch(changeAuthState(value)),
    signOut: () => dispatch(signOut()),
    setLocale: (value) => dispatch(setLocale(value)) 
  })
)(TitleBar)
