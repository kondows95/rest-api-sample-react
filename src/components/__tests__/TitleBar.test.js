import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import TitleBar from '../TitleBar';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})
const fetchAuthedUser = jest.fn(() => {
  let user = [{ id: 1, emali: "a@gmail.com" }];
});
const changeAuthState = jest.fn(() => {
  let auth = ture;
});
const signOut = jest.fn(()=>{
  let auth=false;
});


describe("Titalbar component", () => {
  it('matches the snapshot', () => {
    const TitalbarSnapshot= renderer.create(
      <Parent>
        <TitleBar
          fetchAuthedUser={fetchAuthedUser}
          changeAuthState={changeAuthState} />
      </Parent>
    ).toJSON();
    expect(TitalbarSnapshot).toMatchSnapshot();
  });
});

describe("Teting titlebar", () => {
  it('Testing Signout button and shoppingcart icon', () => {
    act(() => {
      ReactDOM.render((
        <Parent>
          <TitleBar
            fetchAuthedUser={fetchAuthedUser}
            changeAuthState={changeAuthState} 
            signOut={signOut}/>
        </Parent>
      ), container);
      const linkArr = document.querySelectorAll('a');
      act(() => {
        linkArr[0].dispatchEvent(new Event('click', { bubbles: true }))
      });
      expect(linkArr[0].getAttribute("href")).toBe("/cart")
    });
  // cannot test becauseof  <Hidden /> but comment <Hidden/> the following testing work
  //   const ButtonArr = document.querySelectorAll('button');
  //   act(() => {
  //     ButtonArr[1].dispatchEvent(new Event('click', { bubbles: true }))
  //   });
  //  expect(signOut).toHaveBeenCalled();
  //  act(() => {
  //   ButtonArr[2].dispatchEvent(new Event('click', { bubbles: true }))
  // });
  // expect(signOut).toHaveBeenCalled();
   });
});