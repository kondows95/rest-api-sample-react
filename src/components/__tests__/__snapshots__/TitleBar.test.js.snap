// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Order component matches the snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-1 MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="Open drawer"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit Mui-disabled MuiIconButton-edgeStart Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={null}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <div
      className="MuiBox-root MuiBox-root-66"
    >
      Sample
    </div>
    <div
      className="MuiBox-root MuiBox-root-67"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiBox-root MuiBox-root-171"
        />
        <a
          href="/cart"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  className="MuiBadge-badge MuiBadge-colorSecondary MuiBadge-invisible"
                />
              </span>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</header>
`;

exports[`Titalbar component matches the snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-1 MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="Open drawer"
      className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit Mui-disabled MuiIconButton-edgeStart Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={null}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </span>
    </button>
    <div
      className="MuiBox-root MuiBox-root-66"
    >
      Sample
    </div>
    <div
      className="MuiBox-root MuiBox-root-67"
    >
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiBox-root MuiBox-root-171"
        />
        <a
          href="/cart"
          onClick={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  className="MuiBadge-badge MuiBadge-colorSecondary MuiBadge-invisible"
                />
              </span>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</header>
`;
